backend/dal.keycloak/Partials/PimsKeycloakUserService.cs

```diff
@@ -129,20 +131,32 @@ public async Task<Entity.User> UpdateUserAsync(Entity.User user)
             // Update PIMS
             _mapper.Map(user, euser);
 
-            // Update Roles.
-            addRoles.ForEach(async r =>
+            // Remove all keycloak groups from user.  // TODO: Only add/remove the ones that should be removed.
+            var userGroups = await _keycloakService.GetUserGroupsAsync(euser.Id);
+            foreach (var group in userGroups)
+            {
+                await _keycloakService.RemoveGroupFromUserAsync(user.Id, group.Id);
```

Comment:
out of curiosity, what happens when one or more of these requests fails? does it break out of the rest of this method and skip the remaining steps, or does it ignore the failure and keep going?

Reply:
It would throw an exception that would stop further requests.  Which would leave PIMS out of sync in the best way possible.  As PIMS will show the original state and failure, so the admin will want to try and save again.  Which would hopefully resolve the issue and sync the user with keycloak fully.